packages/aws-rfdk/lib/lambdas/nodejs/sep-configuration/handler.ts

```diff
@@ -0,0 +1,288 @@
+          if ('BlockDeviceMappings' in launchSpecification) {
+            launchSpecification.BlockDeviceMappings.map((blockDeviceMapping: any) => {
+              if ('noDevice' in blockDeviceMapping) {
+                blockDeviceMapping.noDevice = this.convertToBoolean(blockDeviceMapping.noDevice);
+              }
+              if ('ebs' in blockDeviceMapping) {
+                if ('deleteOnTermination' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.deleteOnTermination = this.convertToBoolean(blockDeviceMapping.ebs.deleteOnTermination);
+                }
+                if ('encrypted' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.encrypted = this.convertToBoolean(blockDeviceMapping.ebs.encrypted);
+                }
+                if ('iops' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.iops = Number.parseInt(blockDeviceMapping.ebs.iops, 10);
+                }
+                if ('volumeSize' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.volumeSize = Number.parseInt(blockDeviceMapping.ebs.volumeSize, 10);
+                }
+              }
+            });
+          }
+        });
+      }
+    }
+    return JSON.stringify(convertedSpotFleetRequestConfigs);
+  }
+
+  private spotFleetPluginConfigsToArray(pluginOptions: any): any {
+    let convertedConfigs = pluginOptions;
+    if ('DeleteInterruptedSlaves' in convertedConfigs) {
+      convertedConfigs.DeleteInterruptedSlaves = this.convertToBoolean(convertedConfigs.DeleteInterruptedSlaves);
+    }
+    if ('DeleteTerminatedSlaves' in convertedConfigs) {
+      convertedConfigs.DeleteTerminatedSlaves = this.convertToBoolean(convertedConfigs.DeleteTerminatedSlaves);
+    }
+    if ('IdleShutdown' in convertedConfigs) {
+      convertedConfigs.IdleShutdown = Number.parseInt(convertedConfigs.IdleShutdown, 10);
+    }
+    if ('ResourceTracker' in convertedConfigs) {
+      convertedConfigs.ResourceTracker = this.convertToBoolean(convertedConfigs.ResourceTracker);
+    }
+    if ('StaggerInstances' in convertedConfigs) {
+      convertedConfigs.StaggerInstances = this.convertToBoolean(convertedConfigs.StaggerInstances);
+    }
+    if ('StrictHardCap' in convertedConfigs) {
+      convertedConfigs.StrictHardCap = this.convertToBoolean(convertedConfigs.StrictHardCap);
+    }
+    if ('UseLocalCredentials' in convertedConfigs) {
+      convertedConfigs.UseLocalCredentials = this.convertToBoolean(convertedConfigs.UseLocalCredentials);
+    }
+
+    let configs = [];
+    for (const [key, value] of Object.entries(pluginOptions)) {
+      if (value !== undefined) {
+        configs.push({
+          Key: key,
+          Value: value,
+        });
+      }
+    }
+    return configs;
+  }
+
+  /**
+   * Retrieve CA certificate data from the Secret with the given ARN.
+   * @param certificateArn
+   */
+  protected async readCertificateData(certificateArn: string): Promise<string> {
```

Comment:
How come this is protected?

Reply:
We had the same function in mongodb code, where it is protected.
I extracted this method to a separate module, so it can be used in both places.
Please let me know what you think.